### ISSUE TITLE
Lora training : Blowing through all RAM (why?)

### ISSUE THREAD

**Issue by PeteSmalls** [bmaltais/kohya_ss#3129]
I have been trying to get Kohya_SS to train a Flux LoRA on system with 64gb system RAM and 4090 for two weeks now. 
It is a set of 100 images 1024x1024 images that have been working fine in Fluxgym.
When training with Fluxgym, it uses 17gb VRAM and only 10gb of the main RAM with none shared.
It works just fine. (other then not being able to use more than 150 files (images/text) in a session.

But I want to try more so I come over to Koyha_SS.
I use the same dataset and settings (repeats 1 / learning rate / etc etc).

It gets set up and looks as I would expect heading into Epoch 1 (17G VRAM and 10G main RAM).

Then it loses it's damn mind.
During epoch 1 it it DESTROYS ram... eats it all with a hockey stick - maxes out and stops training.

It will do the same if a clean slate and just use 2 images.
chomps RAM with no regard.

I have been through all the setting a ton of times.
I have watch youtube vids and read forum posts... 
I have tried:
- caching and not caching
- Batch 4 (like I do with fluxgym) and batch 1
- High vram, Low vram, both, neither
- Memory efficient attenuation / no Memory efficient attenuation
- A variety of FP16 / BF16 / etc etc.
- Adam8, no adam 8
- Different schedulers.
I have:
- Created new fresh environs
- hand installed CUDA
- Direct installed bitsandbytes and verious supporting versions.

I am clearly missing a setting someplace... there is no WAY it needs over 24gb vram AND 64gb system memory to run a flux LoRA with 100 1024x1024 images.

Any help would be honestly appreciated.
I don't even need of the project anymore - it has become a matter of principle. 
:)

**Comment by b-fission** [comment]
Try enabling "Gradient checkpointing" and "Cache Text Encoder Outputs" since those are the big ones.
